backend/typescript/models/testSession.model.ts

```diff
@@ -0,0 +1,81 @@
+import mongoose, { Schema, Document } from "mongoose";
+
+/**
+ * This interface holds information about the result of a single student
+ * on a test
+ */
+export interface Result {
+  /** the name of the student */
+  student: string;
+  /** the score of the student */
+  score: number;
+  /**
+   * a list corresponding to the question list with each fielding indicating
+   * whether the student got the question right or not
+   */
+  breakdown: [boolean];
+}
+
+const ResultSchema: Schema = new Schema({
+  student: {
+    type: String,
+    required: true,
+  },
+  score: {
+    type: Number,
+    required: true,
+  },
+  breakdown: {
+    type: [Boolean],
+    required: true,
+  },
+});
+
+/**
+ * This document contains information about a single test session
+ */
+export interface TestSession extends Document {
+  /** the unique identifier of the test session */
+  id: string;
```

Comment:
Why do we explicitly add a unique identifier here in `TestSession` collection, but not in other collections? i.e `Users`, `Schools`, `Tests`. Is a unique identifier auto-generated, or do we have to add it as a field? If not automatic, maybe we should be adding to other collections as well.